Source/WebGPU/WebGPU/SwapChain.h

```diff
@@ -25,32 +25,44 @@
 
 #pragma once
 
+#import "Device.h"
 #import <wtf/FastMalloc.h>
+#import <wtf/MachSendRight.h>
 #import <wtf/Ref.h>
 #import <wtf/RefCounted.h>
+#import <wtf/RetainPtr.h>
 
 struct WGPUSwapChainImpl {
 };
 
+struct WGPUSurfaceDescriptor;
+
 namespace WebGPU {
 
-class TextureView;
+class Surface;
 
 class SwapChain : public WGPUSwapChainImpl, public RefCounted<SwapChain> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    static Ref<SwapChain> create()
+    static Ref<SwapChain> create(WGPUSurface surface, const WGPUSwapChainDescriptor& descriptor, Device& device)
     {
-        return adoptRef(*new SwapChain());
+        return adoptRef(*new SwapChain(surface, descriptor, device));
     }
 
     ~SwapChain();
 
-    TextureView* getCurrentTextureView();
     void present();
 
+    int width() const;
+    int height() const;
+
 private:
-    SwapChain();
+    SwapChain(WGPUSurface, const WGPUSwapChainDescriptor&, Device&);
+
+    const Ref<Device> m_device;
+    WGPUSwapChainDescriptor m_descriptor;
+    WGPUSurface m_displayBuffer;
+    WGPUSurface m_backBuffer;
```

Comment:
So the drawing buffer is likely a IOSurface, which gets submitted to the WGPUSurface upon present.

The nomenclature should probably be either of these, but consistently:
- front buffer - back buffer
- display buffer - drawing buffer

Swapchain has:
- a drawing buffer
- free buffers
- surface

Surface has:
- display buffer